### ISSUE TITLE
BOINC Manager interface should not fully block interface when connecting with client

### ISSUE THREAD

**Issue by AustinConlon** [BOINC/boinc#4048]
At the very least the close/minimize/maximize buttons at the top of the window should be clickable.

![Screen Shot 2020-10-04 at 2 36 31 PM](https://user-images.githubusercontent.com/1305957/95033188-21521d00-0672-11eb-811c-b18534b07071.png)

**Comment by CharlieFenton** [comment]
Just to clarify, the "Communicating with client" alert is different from "Connecting to client." If you see that alert, the Manager is already connected to the client, but has been waiting over 1.5 seconds for the client to respond to a _Demand RPC_ request. A _Demand RPC_ is one in which the Manager cannot proceed until it receives the response from the client, and so the Manager's event processing must be suspended. 

I have edited the title of this PR accordingly.

It might be possible to implement this RPC by adding those menu items to `CBOINCGUIApp::FilterEvent(wxEvent &event)`, but I don't think so since that is a modal dialog. See [this documentation](https://docs.wxwidgets.org/3.1.0/classwx_event_filter.html).

Since the Manager's event processing must be suspended, I think it might require a major rewrite of the Manager logic to allow handling of any menu items, which is why I implemented this as a modal dialog. Ideally, this dialog should never appear, but we know from experience that is not the reality. 

I retrofitted the Manager years ago to allow for asynchronous RPCs to minimize the situations in which the manager becomes unresponsive to the user, but due to the Manager's basic design I could not find a way to allow the Manager to remain responsive while waiting for certainRPCs to complete. 

The best I could come up with for that situation is to display the dreaded "Communicating with client" dialog to let the user know why the Manager was not responding, and to allow bailing out by providing the "Exit BOINC Manager" button ("Quit BOINC Manager" on Macs) and  "Cancel" button on the dialog. The Cancel option cancels that one RPC, which sometimes resolves the problem but can produce unpredictable results. Unfortunately, the same situation quickly occurs again with another RPC after the cancel.

I called these _Demand RPCs_ or synchronous RPCs; the others I called asynchronous. You can see  these definitions in `enum ASYNC_RPC_TYPE` in _clientgui/AsyncRPC.h_. 

The logic to retrofit the Manager for asynchronous RPCs was quite complex and tricky to prevent problems from race conditions, etc. Changing the "Communicating with client" dialog to a modeless dialog to enable menu items might be possible, but would require extra code to disable all other events and would need to be done very carefully to avoid accidentally breaking things.

**Comment by CharlieFenton** [comment]
One could argue that the issue is really caused by the client, which is not asynchronous. If the client is waiting for an operation to complete, such as a file operation or an RPC with a project server, it blocks and can't process an RPC from the Manager.

**Comment by CharlieFenton** [comment]
Not a real solution, but here is a workaround when running on MacOS: control-click on the BOINC Manager icon in the Dock, then select _Hide_ from the popup menu.

**Comment by AenBleidd** [comment]
This feature request in nice to have and looks interesting but it requires a huge work to make both client and Manager asynchronous.

**Comment by RichardHaselgrove** [comment]
Could our Mac friends identify exactly what (in the commonest cases) is causing the client to become unresponsive? My guess might be that it's waiting for a delayed response from a project server? That could be checked from the Event Log, when it comes back to life.

**Comment by CharlieFenton** [comment]
> My guess might be that it's waiting for a delayed response from a project server?

@RichardHaselgrove: That is one likely scenario, as I wrote in [this comment above](https://github.com/BOINC/boinc/issues/4048#issuecomment-703498530), but there are probably other situations in which the client is busy for more than 1.5 seconds and so hasn't gotten around to responding to an RPC from the Manager.

**Comment by elfring** [comment]
:crystal_ball: I became curious on further improvements also in this software area.

:thought_balloon: Can any issues be better explained around the message “Communicating with BOINC client”?

**Comment by davidpanderson** [comment]
BTW, the client continues to process GUI RPCs
- while waiting for scheduler and account manager RPC responses
- while doing large file copies or unzip/verify
In other words, it's as asynchronous as it can be given the single-thread/select() design.

Things that might trigger 'communicating with client':
starting a huge program with CreateProcess()
writing a big client_state.xml when the disk is busy
... what else?  I may be able to fix some things

**Comment by CharlieFenton** [comment]
> what else? I may be able to fix some things

When it communicates with project servers, does it do so asynchronously? Are there any situations where it might hang waiting for external communication (i.e., over the Internet)?

Could a very large computer (perhaps with a very large number of CPU cores) have so many tasks that some of the RPCs take a long time to transfer (such as get_results, get_state)?

**Comment by elfring** [comment]
>... what else? …

:thought_balloon: I wonder for a few days (also with the current BOINC software version) which circumstances do hinder to present my BOINC project selection so that I can contribute some data processing resources on demand.

**Comment by CharlieFenton** [comment]
> Can it be determined what the “BOINC client” is trying to achieve while it should respond also to a “Demand RPC” from the BOINC manager user interface?

No. Since the client is not responding, there is no way to get information from the client. The "Communicating with client" alert is the best we can do because that is all the information the GUI (i.e. the BOINC Manager) has.

**Comment by elfring** [comment]
>Since the client is not responding, there is no way to get information from the client.

:crystal_ball: Will any other software debugging approaches become relevant then?


>The "Communicating with client" alert is the best we can do because that is all the information the GUI (i.e. the BOINC Manager) has.

:thought_balloon: Can additional information be displayed for the “Demand RPC” which should get a response anyhow?

**Comment by AenBleidd** [comment]
> >Since the client is not responding, there is no way to get information from the client.
> 
> 
> 
> :crystal_ball: Will any other software debugging approaches become relevant then?
> 
> 
> 
> 
> 
> >The "Communicating with client" alert is the best we can do because that is all the information the GUI (i.e. the BOINC Manager) has.
> 
> 
> 
> :thought_balloon: Can additional information be displayed for the “Demand RPC” which should get a response anyhow?

Why this information should be shown to the user? This is a low-level information that is clear for developers only, and will be just a set of random words for an average user.

**Comment by elfring** [comment]
>Why this information should be shown to the user?

I hope that confusion can be reduced further about undesirable software behaviour. :thinking:


>This is a low-level information that is clear for developers only, and will be just a set of random words for an average user.

:thought_balloon: I imagine that additional information would help to find solutions in easier ways.

**Comment by RichardHaselgrove** [comment]
> Why this information should be shown to the user?

The volunteer community - project 'Help desk experts' - can guide other users in the relevance, use, and interpretation of the specialist debug Event Log flags, as and when the need arises.

Of course, if the client is unresponsive, new RPC events can't be added without a client restart - but if a user suffers these events repeatedly, monitoring can be set up in advance.

**Comment by AenBleidd** [comment]
Ok. Looks like currently this discussion goes in the wrong direction.
To clarify this a little bit.
We have two types of RPCs: blocking and non-blocking.
We (maintainers) should identify what RPC are blocking and are there any ways to make them non-blocking and make the UI more responsive.
We have an 'RPC debug' flag that will show all of the in the log. If blocking RPC is sent - then the user has no way to check the log (at least from the GUI) and understand what is going on.
Probably we should find a better way of handling such cases and handle all of the RPCs in the background thread instead of the GUI thread.
But definitely showing debug information on the GUI window is not a good solution.

**Comment by davidpanderson** [comment]
There are two approaches.  One is to make the Manager responsive even when GUI RPCs are taking a long time.  This led to the blocking/non-blocking distinction.  We spent a lot of time on this and I'm not sure much else can be done.

The 2nd is to make GUI RPCs fast, regardless of what else the client is doing.  This may be possible, but I need to understand why some RPCs are slow.  Windows developers can help.  Run the client under the debugger.  When you get a "communicating with client" message, quickly break the client, see where the main thread is, and send me the stack trace (or post it here).

**Comment by elfring** [comment]
>But definitely showing debug information on the GUI window is not a good solution.

:thought_balloon: What does hinder to present further helpful information about blocking RPCs?

**Comment by elfring** [comment]
>The 2nd is to make GUI RPCs fast, regardless of what else the client is doing.

Are such remote procedure calls put into work queues for background execution?


>Windows developers can help.

Developers with other operating systems can eventually also help more.


>Run the client under the debugger.

:thought_balloon: Can the BOINC client still react to data requests by other commands?

**Comment by elfring** [comment]
>…, but I need to understand why some RPCs are slow.

:thought_balloon: Would you get into the mood to take another look at data processing and corresponding function execution statistics?

**Comment by CharlieFenton** [comment]
@elfring As @davidpanderson wrote, we already spent a great deal of time to minimize the times this happens.

Please explain why you are so concerned about this. For most people, this occurs very rarely and for only a very short  duration; are you seeing the "Communicating with client" alert frequently? Is it blocking the Manager for more than a short time? If the answer to these questions is yes, then please give us details about your BOINC configuration, including the OS, your hardware and which projects you are running, so we have a way to try to reproduce it.

We have very few volunteer developers working on BOINC. Please explain why we should prioritize this over the many other tasks that need our attention.

**Comment by elfring** [comment]
>…; are you seeing the "Communicating with client" alert frequently?

Yes. ‒ “Permanently” for a while on my openSUSE Tumbleweed system.

:eyes: I determined by [extending the constructor of the class “`AsyncRPCDlg`”](https://github.com/BOINC/boinc/blob/d8e4df2f69da22a44574b617d0a95ccf024a4cd9/clientgui/AsyncRPC.cpp#L1076 "Update candidate") that I stumbled on difficulties according to [the enumeration element “`RPC_AUTHORIZE`”](https://github.com/BOINC/boinc/blob/d8e4df2f69da22a44574b617d0a95ccf024a4cd9/clientgui/AsyncRPC.h#L88 "Background information: enum RPC_SELECTOR") once more.

:crystal_ball: How can authorisation issues with BOINC projects be resolved better in my case?

**Comment by AenBleidd** [comment]
@elfring, please enable these event log flags, wait for the next hang, go to the BOINC Data directory, locate `stdoutdae.txt` file and post its content here
![image](https://github.com/BOINC/boinc/assets/3234209/be5bccb5-3362-4a0f-9329-dd9b77f36fb3)
This will help us to determine what exactly if going wrong with your client

**Comment by elfring** [comment]
>please enable these event log flags, …

[I can construct the file “cc_config.xml” as requested](https://boinc.berkeley.edu/wiki/Client_configuration#Logging_flags "Logging flags").


>wait for the next hang,

```
Markus_Elfring@Sonne:/var/lib/boinc> boincmgr
```

* I can take another look at information in my extended “AsyncRPCDlg”.
* I can press also the exit button there.
* But I observe that the program “boincmgr” is not stopped so far.
  Thus I stop it explicitly by the application “system monitor”.


>…, locate stdoutdae.txt file

This suggestion should be reconsidered.


>and post its content here

Where would you like to see standard output data on a Linux execution environment here? :thinking:

**Comment by AenBleidd** [comment]
@elfring, I don't need to see the standard output data. I need the logs from the file I mentioned above.
You can play with your extended class as much as you want, but this doesn't help us.
I'm asking you for the logs from your client. Please provide them. You can either put them as a text in the reply or just attach that file, both ways are ok to me.

**Comment by elfring** [comment]
>I don't need to see the standard output data.

Your suggestion could be interpreted in other ways according to published information.
:eyes: https://boinc.berkeley.edu/wiki/Client_configuration

>I need the logs from the file I mentioned above.

I got the impression that the desired log file is not written during my tests here so far.


>You can play with your extended class as much as you want, but this doesn't help us.

I can display additional data (which can be passed as constructor parameters for my needs) in the mentioned message box.

**Comment by AenBleidd** [comment]
> I got the impression that the desired log file is not written during my tests here so far.

Please be sure to enable debug flags I mentioned above. If you have everything disabled - you'll definitely see very few information in that file

**Comment by elfring** [comment]
I would appreciate if I could offer more information also according to the following file content.
```
Sonne:/var/lib/boinc # ls -l cc_config.xml && cat cc_config.xml
-rw-r--r-- 1 boinc boinc 213 Sep 20 15:35 cc_config.xml
<cc_config>
<log_flags>
<file_xfer>1</file_xfer>
<gui_rpc_debug>1</gui_rpc_debug>
<http_debug>1</http_debug>
<http_xfer_debug>1</http_xfer_debug>
<sched_ops>1</sched_ops>
<task>1</task>
</log_flags>
</cc_config>
```

:thought_balloon: Unfortunately, it seems that some parts of the discussed software are still not working as expected at the moment.

**Comment by RichardHaselgrove** [comment]
> I got the impression that the desired log file is not written during my tests here so far.

`stdoutdae.txt` is only one possible location for BOINC's permanent record of previous Event Logs. It depends how you start the client under your particular OS. Most modern Linux implementations launch the client as a systemd service, and in those cases, the event log is held in the system Journal. You mentioned "openSUSE Tumbleweed" a little while ago: I'm not familiar with that, but some boinc documentation suggests the use of a `-redirectio` switch if starting the client from the command line - e.g. https://github.com/BOINC/boinc/wiki/CoreClient

**Comment by elfring** [comment]
I took another look also at the system configuration of my local service “boinc-client-variant”.
The corresponding log data point the information “Sep 20 … Sonne boinc[…]: Authorization required, but no authorization protocol specified” out for further considerations.

**Comment by AenBleidd** [comment]
@elfring, https://bbs.archlinux.org/viewtopic.php?id=272491

**Comment by AenBleidd** [comment]
Also: https://boinc.berkeley.edu/dev/forum_thread.php?id=14249

**Comment by elfring** [comment]
:crystal_ball: How many developers and code reviewers would get into the mood to explain the application of [the enumeration element “`RPC_AUTHORIZE`”](https://github.com/BOINC/boinc/blob/d8e4df2f69da22a44574b617d0a95ccf024a4cd9/clientgui/AsyncRPC.h#L88 "Background information: enum RPC_SELECTOR") better?

Which passed values do you observe in [the member variable “`which_rpc`” of the data structure “`ASYNC_RPC_REQUEST`”](https://github.com/BOINC/boinc/blob/ed69f05b396fcd09007181bb72ea61927f182bb0/clientgui/AsyncRPC.h#L191 "Background information: struct ASYNC_RPC_REQUEST") (during reported graphical user interface blockages)? :thinking:

**Comment by AenBleidd** [comment]
@elfring, you should be able to find this information in our documentation: https://github.com/BOINC/boinc/wiki/GuiRpcProtocol

**Comment by elfring** [comment]
:thought_balloon: I am unsure if the enumeration element “RPC_AUTHORIZE” is mapped to a “[GUI RPC Protocol](https://github.com/BOINC/boinc/wiki/GuiRpcProtocol "Wiki article")” item here.

**Comment by CharlieFenton** [comment]
> I determined by [extending the constructor of the class “AsyncRPCDlg”](https://github.com/BOINC/boinc/blob/d8e4df2f69da22a44574b617d0a95ccf024a4cd9/clientgui/AsyncRPC.cpp#L1076) that I stumbled on difficulties according to [the enumeration element “RPC_AUTHORIZE”](https://github.com/BOINC/boinc/blob/d8e4df2f69da22a44574b617d0a95ccf024a4cd9/clientgui/AsyncRPC.h#L88) once more.

 The Manager calls _RPC_AUTHORIZE_ (actually _rpc.authorize_) only once when first connecting to the client, to confirm that the Manager is authorized to communicate with the client. This is the first RPC the Manager sends to the client, so If the client is not responding to this it means the client has _never_ responded to the Manager.

**Comment by elfring** [comment]
:thought_balloon: Can such an authorisation check request be logged also by the BOINC client (RPC service) so that this information would appear in the systemd journal?

**Comment by AenBleidd** [comment]
@elfring, if the corresponding flags (shown on my screenshot above) are set and the client is running and responsive - yes, this RPC will also be logged

**Comment by elfring** [comment]
:thought_balloon: I would appreciate if any contributors can point the source code place out which should trigger an expected log message.

I would like to filter better on helpful messages for this issue than [related information](https://github.com/BOINC/boinc/issues/5374 "Fix software weaknesses around the message “Authorization required, but no authorization protocol specified”") which might belong to other remaining open issues.

**Comment by elfring** [comment]
>Why you can't just share all logs?

* I got doubts that [a corresponding logging configuration](https://github.com/BOINC/boinc/issues/4048#issuecomment-1727902151 "Background information from 2023-09-20") achieved desirable effects.
* The questionable frequency of the message “Authorization required, but no authorization protocol specified” makes analyses of undesirable software behaviour more challenging and complex.

**Comment by AenBleidd** [comment]
If you have doubts - share your logs. There will be said, what configuration was loaded, what log flags are used, were your configuration file correct or not, etc.
Also, all unrelevant information could be filtered by us, this is not as issue.
Without logs at all is much harder to determine what's wrong on your side.

**Comment by elfring** [comment]
>…, what configuration was loaded, what log flags are used, were your configuration file correct …

Please provide more constructive comments on the linked file “cc_config.xml”.


>Also, all unrelevant information could be filtered by us, this is not as issue.

:eyes: I am looking also for better data filter capabilities here.

:thought_balloon: Which functions should achieve helpful data output then?

**Comment by AenBleidd** [comment]
Attached file looks correct. You need to restart the BOINC client if you change this file manually.
After you do that - please provide ALL logs from the BOINC client.
Thank you

**Comment by elfring** [comment]
>After you do that - please provide ALL logs from the BOINC client.

I got the impression that the data size became too big.
Thus I would prefer to concentrate on improvements for relevant log messages.
:crystal_ball: Which data would you expect there so far?

**Comment by CharlieFenton** [comment]
> The questionable frequency of the message “Authorization required, but no authorization protocol specified” makes analyses of undesirable software behaviour more challenging and complex.

A Google search for that error message turns up many discussions of it. Apparently it is associated with X Windows on Linux. [This](https://boinc.berkeley.edu/dev/forum_thread.php?id=14249) is one of the results from my Google search.

**Comment by AenBleidd** [comment]
@elfring, this piece of log doesn't contain any lines related to the RPC handling. Are there any other log lines with the `[gui_rpc]` prefix or these two are the only one?

**Comment by elfring** [comment]
>Are there any other log lines with the `[gui_rpc]` prefix

No (for this test approach so far).


>or these two are the only one?

Yes (at the moment).


Would you like to fiddle with any diff command variations for our evolving clarification of more desirable software behaviour? :thinking:

**Comment by elfring** [comment]
>This is the first RPC the Manager sends to the client, so If the client is not responding to this it means the client has _never_ responded to the Manager.

:crystal_ball: Will such a view trigger growing development interests?

**Comment by elfring** [comment]
:crystal_ball: Do you find another test result interesting for further clarifications?
```
Sonne:/var/lib/boinc # /usr/local/bin/boinccmd --get_tasks
Authorization failure: -102
```

**Comment by RichardHaselgrove** [comment]
> Sonne:/var/lib/boinc # /usr/local/bin/boinccmd --get_tasks
> Authorization failure: -102

I strongly believe that one is due to a failure to pass the RPC password, stored in gui_rpc_auth.cfg, to the client. For testing, pass it on the command line as shown in boinccmd documentation.

**Comment by elfring** [comment]
:crystal_ball: Would you like to add further ideas to another test result?
```
Sonne:/var/lib/boinc # time /usr/local/bin/boinccmd --passwd $(cat gui_rpc_auth.cfg) --client_version
Authorization failure: -102

real    0m30.655s
user    0m0.018s
sys     0m0.017s
```

**Comment by CharlieFenton** [comment]
>  strongly believe that one is due to a failure to pass the RPC password, stored in gui_rpc_auth.cfg, to the client. For testing, pass it on the command line as shown in boinccmd documentation.

If you look at lib//error_numbers.h you will see `#define ERR_READ            -102` which indicates it is able to read the _gui_rpc_auth.cfg_ file in the BOINC data directory. Thus passing `--passwd $(cat gui_rpc_auth.cfg` is not the same as what @RichardHaselgrove suggested you do. You need to pass the actual value in the command line for the test he suggested.

This error is almost always due to read permissions. This is a very common issue which the volunteer help staff on the BOINC forums can easily address. 


As I wrote before, a Google search for the message “Authorization required, but no authorization protocol specified" turns up many discussions of it. Apparently it is associated with X Windows on Linux. [This](https://boinc.berkeley.edu/dev/forum_thread.php?id=14249) is one of the results from my Google search.

For the common types of errors you are seeing, the BOINC forums are the appropriate place to seek help, rather than wasting the developers' time as you have been doing by your constant arguing and refusal to accept requests and advice from @AenBleidd.

**Comment by elfring** [comment]
>Thus passing `--passwd $(cat gui_rpc_auth.cfg)` is not the same as what @RichardHaselgrove suggested you do. You need to pass the actual value in the command line for the test he suggested.

Another test result demonstrates that [more desirable data processing effects](https://boinc.berkeley.edu/wiki/Boinccmd_tool#State_queries "State queries for BOINC") could also not be achieved so far after I copied the password directly (and not the shown horizontal ellipsis as a placeholder) from [the mentioned file](https://boinc.berkeley.edu/wiki/Controlling_BOINC_remotely#Access_control_for_GUI_RPC "Access control for GUI RPC").
```
Sonne:/var/lib/boinc # time /usr/local/bin/boinccmd --passwd '…' --client_version
Authorization failure: -102

real    0m31.702s
user    0m0.003s
sys     0m0.009s
```


>Apparently it is associated with X Windows on Linux.

[It was indicated finally](https://github.com/BOINC/boinc/issues/5374#issuecomment-1737285140 "Feedback by Vitalii Koshura from 2023-09-27") that this message would be connected to an idle detection technique.

The software can be improved further accordingly.

**Comment by AenBleidd** [comment]
> The software can be improved further accordingly.

This is something we are doing on daily basis.
Unfortunately, currently this discussion is completely out of the topic of this particular issue.
All the other messages not related to the original issue will be deleted.

@elfring, please search for the solutions for the issues you have on the BOINC forums: https://boinc.berkeley.edu/forum_index.php

**Comment by elfring** [comment]
>You need to pass the actual value in the command line for the test he suggested.

1. I can also rename the file “gui_rpc_auth.cfg” for another system check.
2. A different password was automatically written by the BOINC software into a new file after the service “boinc-client-variant” was restarted.
3. The information “Authorization failure: -102” was displayed again.

**Comment by RichardHaselgrove** [comment]
As @CharlieFenton said, error -102 is a file read error.

That points to a problem with the file location, ownership, or permissions. Failure to comply with BOINC's security model is a user or distribution problem, not a coding issue.

**Comment by elfring** [comment]
>That points to a problem with the file location, ownership, or permissions.

I can also temporarily try further checks out with a known file.
```
Sonne:/var/lib/boinc # ls -l gui_rpc_auth.cfg
-rw-r--r-- 1 boinc boinc 32 Sep 29 10:09 gui_rpc_auth.cfg
```

**Comment by AenBleidd** [comment]
@elfring, have you added your user to the 'boinc' group?

**Comment by elfring** [comment]
Yes.

:eyes: See also:
```
Markus_Elfring@Sonne:~> groups
Markus_Elfring postgres boinc
```

**Comment by AenBleidd** [comment]
@elfring, then you should ask for the further assistance on the BOInC forum.
As I already told - the issue you have doesn't related to the original issue reported in this topic.
So please, stop flooding here.

**Comment by AenBleidd** [comment]
@elfring, -> https://boinc.berkeley.edu/forum_index.php

**Comment by elfring** [comment]
>Failure to comply with BOINC's security model is a user or distribution problem, not a coding issue.

The exception handling can be adjusted a bit like the following.
```diff
diff --git a/lib/gui_rpc_client.cpp b/lib/gui_rpc_client.cpp
index 21c412cc04..af59d9c387 100644
--- a/lib/gui_rpc_client.cpp
+++ b/lib/gui_rpc_client.cpp
@@ -326,7 +326,14 @@ int RPC_CLIENT::get_reply(char*& mbuf) {
     mf.puts("");    // make sure buffer is non-NULL
     while (1) {
         n = recv(sock, buf, 8192, 0);
-        if (n <= 0) return ERR_READ;
+        if (n == 0)
+            goto return_error_code;
+        if (n < 0) {
+            perror("recv");
+return_error_code:
+            return ERR_READ;
+        }
+
         buf[n]=0;
         mf.puts(buf);
         if (strchr(buf, '\003')) break;
```


Another bit of information could become available for [further development considerations](https://github.com/BOINC/boinc/blob/8b5334d5bd759dc85cf9be6526af5fcdfcc0b112/lib/gui_rpc_client.cpp#L319 "Update candidate: RPC_CLIENT::get_reply()").
```
Markus_Elfring@Sonne:/var/lib/boinc> /usr/local/bin/boinccmd --client_version
recv: Resource temporarily unavailable
Authorization failure: -102
```

**Comment by AenBleidd** [comment]
@elfring, this is a library. Library doesn't print errors to the std.
Please stop flooding tickets with a Job-relevant information.
If you continue doing this - I'll have to block you here. Is it clear?

**Comment by elfring** [comment]
>If you continue doing this - I'll have to block you here.

* Would you like to overlook the presented error information “`recv: Resource temporarily unavailable`” (in this issue tracker)?
* Do you insist to clarify this detail only by the forum interface?

**Comment by AenBleidd** [comment]
> Would you like to overlook the presented error information “recv: Resource temporarily unavailable” (in this issue tracker)?

No.

> Do you insist to clarify this detail only by the forum interface?

Yes.

**Comment by RichardHaselgrove** [comment]
> `recv: Resource temporarily unavailable`

This information is totally useless to the end user:
* It doesn't identify which resource is unavailable
* It doesn't indicate why it may be unavailable
* It doesn't give any guidance to how the user could resolve it

I agree with @AenBleidd: take it to the forums, until you understand the context better.

**Comment by elfring** [comment]
>This information is totally useless to the end user:

:thought_balloon: The shown error information could be helpful for system testers and software developers (who would like to care a bit more for [implementation details also in the function “RPC_CLIENT::get_reply”](https://github.com/BOINC/boinc/blob/8b5334d5bd759dc85cf9be6526af5fcdfcc0b112/lib/gui_rpc_client.cpp#L319 "Update candidate")), couldn't it?

**Comment by AenBleidd** [comment]
@elfring, we have error codes for that. And that's completely enough: if resource is unavailable, we don't care why exactly it's unavailable, that is why we have a general error code.
For the solution - please go to the forum.

**Comment by elfring** [comment]
>And that's completely enough: if resource is unavailable, we don't care why exactly it's unavailable, that is why we have a general error code.

:crystal_ball: Would any contributors like to reconsider such a view (also with the help of the following data representation)?
```
Markus_Elfring@Sonne:/var/lib/boinc> strace --trace=%network /usr/local/bin/boinccmd --client_version
socket(AF_INET, SOCK_STREAM, IPPROTO_IP) = 3
setsockopt(3, SOL_SOCKET, SO_RCVTIMEO_OLD, "\36\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0", 16) = 0
connect(3, {sa_family=AF_INET, sin_port=htons(31416), sin_addr=inet_addr("127.0.0.1")}, 16) = 0
sendto(3, "<boinc_gui_rpc_request>\n<auth1/>"..., 59, 0, NULL, 0) = 59
recvfrom(3, 0x7ffda97ba150, 8192, 0, NULL, NULL) = -1 EAGAIN (Die Ressource ist zur Zeit nicht verfügbar)
recv: Resource temporarily unavailable
Authorization failure: -102
+++ exited with 1 +++
```

**Comment by AenBleidd** [comment]
I had to block @elfring from this organization because they continue posting unrelevant messages even after several asks to not do that.

**Comment by Ageless93** [comment]
> I had to block @elfring from this organization because they continue posting unrelevant messages even after several asks to not do that.

Thanks for that. Now he's messing up the forums. Not sure what to do about him yet, maybe he'll go away if he's being ignored. ;)
